test/fail_compilation/test20023.d

```diff
@@ -0,0 +1,16 @@
+// REQUIRED_ARGS: -preview=dip1000 -preview=dip1008 -Ifail_compilation/extra-files
+// https://issues.dlang.org/show_bug.cgi?id=20023
+/*
+TEST_OUTPUT:
+---
+fail_compilation/extra-files/test20023b.d(8): Error: scope variable `e` may not be returned
+fail_compilation/test20023.d(15): Error: template instance `test20023b.threw!()` error instantiating
+---
+*/
+import test20023b;
```

Comment:
Move `test20023b.d` to `imports` and reference it as `imports.test20023b`.
Then remove the additional import path defined above.